packages/components/app/styles/components/dropdown.scss

```diff
@@ -408,8 +416,86 @@ $hds-dropdown-toggle-border-radius: 5px;
   }
 }
 
+// HDS::DROPDOWN::LIST-ITEM::TITLE
+
 .hds-dropdown-list-item--title {
   padding: 10px 16px 4px;
   color: var(--token-color-foreground-strong);
 }
 
+// HDS::DROPDOWN::LIST-ITEM::CHECKMARK
+
+.hds-dropdown-list-item--checkmark {
+  a.hds-dropdown-list-item__interactive,
+  button.hds-dropdown-list-item__interactive {
+    // leave room for the checkmark to align the result count on non-selected items
+    padding-right: 25px; // 9px default + 16px for the 'check' FlightIcon
```

Comment:
Maybe you could use a parent container, with its own intrinsic size, and then conditionally render the icon inside it, so that you don't need to use extra padding (and calculations) to achieve the desired layout

Reply:
Followed the suggestion and refactored to use a fixed-width container, that conditionally renders the checkmark icon in [35201dd](https://github.com/hashicorp/design-system/pull/1185/commits/35201dd1eb17d523f19e16967eeac30e8dfe3875)